### DISCUSSION TITLE
Topaz extract equivalent of separation radius?

### DISCUSSION THREAD

**Discussion by marinegor** [tbepler/topaz#115] [Q&A]
Hi everyone, I was using Topaz integration with CryoSPARC and just noticed that the particles after Topaz extract overlap a lot. Is there a way to control somehow? I was looking for something like "overlap radius" in CryoSPARC's "Template picker", but couldn't find any. Any suggestions?

**Comment by tbepler** [comment]
I think the "radius" parameter of "topaz extract" is what you're looking for.

**Reply by marinegor** [reply]
Thanks, this helps a bit -- the `topaz extract --help` is indeed more helpful than a CryoSPARC's annotation.

But still, how do I control the overlap? If I e.g. want my particles to not overlap at all, is there a way to control it directly?
I've seen an [answer](https://discuss.cryosparc.com/t/topaz-picking-radius/4057) at the CryoSPARC discussion forum mentioning that there is no direct way to control overlapping -- I wonder, if it's still the case.

**Reply by tbepler** [reply]
This parameter controls the overlap. If you set radius to 5 (for example) when running topaz extract, then the extracted particles centers cannot be closer than 5 pixels.

**Reply by marinegor** [reply]
In my hands (=using CryoSPARC's 3.2 interface) it does not influence the number of extracted particles. Using following line:

```Starting extraction by running command /opt/conda3/envs/topaz/bin/topaz extract --radius 7 --threshold -6 --up-scale 4 --assignment-radius -1 --min-radius 300 --max-radius 400 --step-radius 5 --num-workers 8 --device 2 --model /data/cryosparc_projects/P3/J619/models/model_epoch030.sav -o /data/cryosparc_projects/P3/J707/topaz_particles_prediction.txt```

When I set `--min-radius 5 --max-radius 100`, I get the same number of particles as with `(300, 400)` pair, which probably shouldn't be the case🤔

**Reply by tbepler** [reply]
The min and max radius parameters don't do anything in the cryoSPARC command. They exist for cross validating the radius parameter. The parameter you want to adjust is just --radius.

**Reply by marinegor** [reply]
Ok, indeed. Thanks a lot!

**Reply by olibclarke** [reply]
> This parameter controls the overlap. If you set radius to 5 (for example) when running topaz extract, then the extracted particles centers cannot be closer than 5 pixels.

Just to double check this - the radius, not 2x the radius, is the minimum distance between particle centers?

**Reply by tbepler** [reply]
That's correct.
